molecularnodes/entities/molecule/base.py

```diff
@@ -60,15 +49,108 @@ def __init__(self, file_path: Union[str, Path, io.BytesIO]):
-        self.file: str
-        self.array: np.ndarray
-        self._frames_collection: str | None
+        self._frames_collection: str | None = None
+        self._code: str | None = None
         self._entity_type = EntityType.MOLECULE
+        super().__init__()
+        self._read(file_path)
+        self.atom_array: AtomArray | AtomArrayStack
+        self._create_object()
+
+    @property
+    def code(self) -> str | None:
+        """
+        Get the code for the molecule.
+        """
+        return self._code
+
+    def _read(self, file_path: str | Path | io.BytesIO) -> None:
+        """
+        Initially open the file, ready to extract the required data.
+
+        Parameters
+        ----------
+        file_path : Union[Path, io.BytesIO]
+            The file path to the file which stores the atomic coordinates.
+        """
+        if isinstance(file_path, io.BytesIO):
+            self._reader = pdbx.PDBXReader(file_path)
+        else:
+            if isinstance(file_path, str):
+                file_path = Path(file_path)
+
+            match file_path.suffix:
+                case ".cif":
+                    self._reader = pdbx.PDBXReader(file_path)
+                case ".bcif":
+                    self._reader = pdbx.PDBXReader(file_path)
+                case ".pdb":
+                    self._reader = pdb.PDBReader(file_path)
+                case ".sdf":
+                    self._reader = sdf.SDFReader(file_path)
+                case ".mol":
+                    self._reader = sdf.SDFReader(file_path)
+                case _:
+                    raise InvalidFileError("The file format is not supported.")
+
+        self.atom_array = self._reader.array.copy()
+        try:
+            self._assemblies = self._reader._assemblies()
+        except InvalidFileError:
+            self._assemblies = ""
+
+    @classmethod
+    def fetch(
+        cls,
+        code: str,
+        format=".bcif",
+        centre: str | None = None,
+        cache=download.CACHE_DIR,
+        database: str = "rcsb",
+    ):
+        """
+        Fetch a molecule from the RCSB database.
+
+        Parameters
+        ----------
+        code : str
+            The PDB ID code of the molecule to fetch.
+        format : str, optional
+            The file format to download. Default is ".bcif".
+        cache : str, optional
+            Path to cache directory. If None, no caching is performed.
+        database : str, optional
+            The database to fetch from. Default is "rcsb".
+
+        Returns
+        -------
+        Molecule
+            A new Molecule instance created from the downloaded data.
+        """
+        file_path = download.download(
+            code=code, format=format, cache=cache, database=database
+        )
+        mol = cls(file_path)
+        mol.object.mn["entity_type"] = "molecule"
+        mol._code = code
+
+        if centre:
+            mol.position + -mol.centroid(centre)
```

Comment:
might want to declare `mol.position` as an attribute at top. 

Otherwise it might be `mol.object.postion`?